src/utils/graphQLClient.tsx

```diff
@@ -56,56 +54,69 @@ export const getFetchOptions = (token?: string) => ({
         },
+      }),
+      authExchange(async (utils) => {
+        return {
+          addAuthToOperation(operation) {
+            if (!currentToken) return operation;
+            return utils.appendHeaders(operation, { Authorization: `Bearer ${currentToken}` });
+          },
 
-        // Determine if the current error is an authentication error.
-        didAuthError: (error) =>
-          error.response.status === 401 ||
-          error.graphQLErrors.some((e) => e.message.includes("Must login")),
+          // Determine if the current error is an authentication error.
+          didAuthError: (error) =>
+            error.response.status === 401 ||
+            error.graphQLErrors.some((e) => e.message.includes("Must login")),
 
-        // If didAuthError returns true, clear the token. Ideally we should refresh the token here.
-        // The operation will be retried automatically.
-        async refreshAuth() {
-          setCurrentToken(null);
-        },
+          // If didAuthError returns true, clear the token. Ideally we should refresh the token here.
+          // The operation will be retried automatically.
+          async refreshAuth() {
+            setCurrentToken(null);
+          },
 
-        // Prevent making a request if we know the token is missing, invalid or expired.
-        // This handler is called repeatedly so we avoid parsing the token each time.
-        willAuthError() {
-          if (!currentToken) return true;
-          try {
-            if (!currentTokenExpiration) {
-              const { exp } = JSON.parse(atob(currentToken.split(".")[1]));
-              currentTokenExpiration = exp;
+          // Prevent making a request if we know the token is missing, invalid or expired.
+          // This handler is called repeatedly so we avoid parsing the token each time.
+          willAuthError() {
+            if (!currentToken) return true;
+            try {
+              if (!currentTokenExpiration) {
+                const { exp } = JSON.parse(atob(currentToken.split(".")[1]));
+                currentTokenExpiration = exp;
+              }
+              return Date.now() / 1000 > (currentTokenExpiration || 0);
+            } catch (e) {
+              return true;
             }
-            return Date.now() / 1000 > (currentTokenExpiration || 0);
-          } catch (e) {
-            return true;
-          }
-        },
-      };
-    }),
-    fetchExchange,
-  ],
-  fetchOptions: getFetchOptions(), // Auth header (token) is handled by authExchange
-});
+          },
+        };
+      }),
+      fetchExchange,
+    ],
+    fetchOptions: getFetchOptions(), // Auth header (token) is handled by authExchange
+    ...options,
+  });
 
-export const GraphQLClientProvider = ({ children }: { children: React.ReactNode }) => {
-  return <Provider value={client}>{children}</Provider>;
+export const GraphQLClientProvider = ({
+  children,
+  value = createClient(),
+  ...rest
```

Comment:
rest has no types

Reply:
I was doing this because of the `key` prop, but I’m not sure it’s necessary to pass that down.

Reply:
You could consider using `FC` from `react for that.